dotnet-desktop-guide/net/winforms/compiler-messages/wfo0003.md

```diff
@@ -0,0 +1,131 @@
+
+  <PropertyGroup>
+    <OutputType>WinExe</OutputType>
+    <TargetFramework>net9.0-windows</TargetFramework>
+    <Nullable>enable</Nullable>
+    <UseWindowsForms>true</UseWindowsForms>
+    <ImplicitUsings>enable</ImplicitUsings>
+
+    <ApplicationHighDpiMode>SystemAware</ApplicationHighDpiMode>
+    <ApplicationVisualStyles>true</ApplicationVisualStyles>
+    <ApplicationUseCompatibleTextRendering>false</ApplicationUseCompatibleTextRendering>
+    <ApplicationDefaultFont>Microsoft Sans Serif, 8.25pt</ApplicationDefaultFont>
+
+  </PropertyGroup>
+
+</Project>
+```
+
+These settings generate the following method:
+
+```csharp
+[CompilerGenerated]
+internal static partial class ApplicationConfiguration
+{
+    public static void Initialize()
+    {
+        global::System.Windows.Forms.Application.EnableVisualStyles();
+        global::System.Windows.Forms.Application.SetCompatibleTextRenderingDefault(false);
+        global::System.Windows.Forms.Application.SetHighDpiMode(HighDpiMode.SystemAware);
+        global::System.Windows.Forms.Application.SetDefaultFont(new Font(new FontFamily("Microsoft Sans Serif"), 8.25f, (FontStyle)0, (GraphicsUnit)3));
+    }
+}
+```
+
+### Using Visual Basic
+
+Visual Basic sets the High DPI mode in two places. The first is in the project properties, which affects the Visual Studio Designer, but doesn't affect the app at run-time.
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>WinExe</OutputType>
+    <TargetFramework>net9.0-windows</TargetFramework>
+    <StartupObject>Sub Main</StartupObject>
+    <UseWindowsForms>true</UseWindowsForms>
+    <MyType>WindowsForms</MyType>
+
+    <ApplicationHighDpiMode>SystemAware</ApplicationHighDpiMode>
+    <ApplicationVisualStyles>false</ApplicationVisualStyles>
+    <ApplicationUseCompatibleTextRendering>false</ApplicationUseCompatibleTextRendering>
+    <ApplicationDefaultFont>Microsoft Sans Serif, 8.25pt</ApplicationDefaultFont>
+
+  </PropertyGroup>
+
+</Project>
+```
+
+The second place is in the Visual Basic Application Framework settings. These are in the project properties page, under **Application** > **Application Framework**. These settings are saved into the _My Project\\Application.myapp_ file or in an application startup event handler.
+
+### Setting the font in Visual Basic
+
+The run-time font, however, isn't configurable in Visual Basic through an app setting or the project file. It must be set in the <xref:Microsoft.VisualBasic.ApplicationServices.WindowsFormsApplicationBase.ApplyApplicationDefaults> event handler.
+
+```vb
+Imports Microsoft.VisualBasic.ApplicationServices
+
+Namespace My
+    Partial Friend Class MyApplication
+        Private Sub MyApplication_ApplyApplicationDefaults(sender As Object, e As ApplyApplicationDefaultsEventArgs) Handles Me.ApplyApplicationDefaults
+            e.Font = New Font("Microsoft Sans Serif", 8.25)
+        End Sub
+    End Class
+End Namespace
+```
```

Comment:
@KlausLoeffelmann This content was in the other wfac010 which seems was swapped to this warning wfo0003. I don't know why the font setting is mentioned here considering this errors shows with the DPI setting?

Reply:
> to this warning wfo0003

Yes. To standardize the Analyzer warning format.

I am not sure, if the numbers then are correct. We phased out the old numbers completely, and replaced them with the new format. The easiest way is to see what pops up in the compiler.

Note that the main application for this is that the _Designer_ already uses the new font settings for Design purposes. But this won't work for VB, if you set the font like described programmatically. You CAN set the font the same way as you do for C# though, but that ONLY effects the Designer at Design time, and has no effect on the runtime. The additional programmatical way is still necessary for runtime adaptation.